include/exadg/operators/finite_element.h

```diff
@@ -0,0 +1,128 @@
+/*  ______________________________________________________________________
+ *
+ *  ExaDG - High-Order Discontinuous Galerkin for the Exa-Scale
+ *
+ *  Copyright (C) 2023 by the ExaDG authors
+ *
+ *  This program is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation, either version 3 of the License, or
+ *  (at your option) any later version.
+ *
+ *  This program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ *  ______________________________________________________________________
+ */
+
+#ifndef INCLUDE_EXADG_OPERATORS_FINITE_ELEMENT_H_
+#define INCLUDE_EXADG_OPERATORS_FINITE_ELEMENT_H_
+
+// deal.II
+#include <deal.II/fe/fe_dgq.h>
+#include <deal.II/fe/fe_q.h>
+#include <deal.II/fe/fe_simplex_p.h>
+#include <deal.II/fe/fe_system.h>
+
+// ExaDG
+#include <exadg/grid/grid_data.h>
+#include <exadg/utilities/exceptions.h>
+
+namespace ExaDG
+{
+/**
+ * Creates and returns a shared pointer to dealii::FiniteElement<dim> depending on the element type,
+ * the type of function space, the number of components of the solution field, and the polynomial
+ * degree of the shape functions. Regarding the function space, the code assumes an H^1 conforming
+ * space (Continuous Galerkin) if is_dg (= L^2-conforming space, Discontinuous Galerkin) is false.
+ */
+template<int dim>
+std::shared_ptr<dealii::FiniteElement<dim>>
+create_finite_element(ElementType const & element_type,
+                      bool const          is_dg,
```

Comment:
We don't have the infrastructure right now, but I would prefer to pass
```suggestion
                      SpatialDiscretization(Type) const          discretization_type,
```
here, rather than `true/false`. When I say we are not there, I mean that we currently define the `SpatialDiscretization` enum only in some module, but not everywhere. Maybe we also do not need to re-introduce this parameter in ExaDG, as deal.II already has an enum with the various conformity settings https://github.com/dealii/dealii/blob/61a8f375f62f7ada3edd1f269e3dce1bd5973ccf/include/deal.II/fe/fe_data.h#L262-L296. We probably do not want to write `dealii::FiniteElementData<dim>::Conformity` but might introduce a shortcut. If we have consensus on this topic, I can introduce that in a follow-up.

Reply:
Yes, let's wait with this and do it somewhere else. Note also that `OperatorBase` and `MultigridPreconditionerBase` use `bool is_dg`. This is why I intentionally followed the current procedure also in the present PR.